### PR TITLE
Minor fix: Run replication_sync tests only if failure compiled in.

### PR DESCRIPTION
**Pull request by neunhoef**

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6090

File: tests/js/server/replication/sync/replication-sync.js [hunk 1]
```diff
@@ -2321,9 +2321,11 @@ function ReplicationNonIncrementalKeyConflict () {
...
 // //////////////////////////////////////////////////////////////////////////////
 
-jsunity.run(ReplicationSuite);
-jsunity.run(ReplicationOtherDBSuite);
-jsunity.run(ReplicationIncrementalKeyConflict);
-jsunity.run(ReplicationNonIncrementalKeyConflict);
+if (internal.debugCanUseFailAt()) {
+  jsunity.run(ReplicationSuite);
+  jsunity.run(ReplicationOtherDBSuite);
+  jsunity.run(ReplicationIncrementalKeyConflict);
```
Comment (mchacki) [inline tests/js/server/replication/sync/replication-sync.js]:
```suggestion
}
jsunity.run(ReplicationIncrementalKeyConflict);
```

File: tests/js/server/replication/sync/replication-sync.js [hunk 2]
```diff
@@ -2321,9 +2321,11 @@ function ReplicationNonIncrementalKeyConflict () {
...
+  jsunity.run(ReplicationNonIncrementalKeyConflict);
```
Comment (mchacki) [inline tests/js/server/replication/sync/replication-sync.js]:
```suggestion
jsunity.run(ReplicationNonIncrementalKeyConflict);
```

File: tests/js/server/replication/sync/replication-sync.js [hunk 3]
```diff
@@ -2321,9 +2321,11 @@ function ReplicationNonIncrementalKeyConflict () {
...
+}
```
Comment (mchacki) [inline tests/js/server/replication/sync/replication-sync.js]:
```suggestion
```

I think excluding only the first two suites is enough.
Reading the file they seem to attempt to check themselves if debugFailAt is actually activated. Which for some reason does not work in all cases.

**Comment by neunhoef** [comment]
Too low prio to get a time slice.

**Pull request closed without merging**